### PR TITLE
#273 throw NoAlertPresentException, NoSuchWindowException, or NoSuchF…

### PR DESCRIPTION
**Pull request by tsukakei**
…rameException instead of TimeoutException

## Proposed changes
Fow now, `switchTo().alert()` throws TimeoutException as well as `switchTo().frame()` and `switchTo().window()` do.
I think these methods should throw `NoAlertPresentException`, `NoSuchFramePresentException`, or `NoSuchWindowPresentException` respectively.

This PR is related to #273.

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by symonk** [comment]
perhaps IT tests around assertsThrownBy to ensure correct exceptions are thrown under the given scenarios, I started on this myself too so thanks for doing this :) I have no time to review it this morning, I need to pop out.  Collaborator(s) will come along shortly with some feedback.

File: src/main/java/com/codeborne/selenide/SelenideTargetLocator.java [hunk 1]
```diff
@@ -62,7 +73,11 @@ public WebElement activeElement() {
 
   @Override
   public Alert alert() {
-    return Wait().until(alertIsPresent());
+    try {
+      return Wait().until(alertIsPresent());
+    } catch (TimeoutException e) {
+      throw new NoAlertPresentException();
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/SelenideTargetLocator.java]: Don't forget to wrap the original exception: `throw new NoAlertPresentException(e);`

File: src/main/java/com/codeborne/selenide/SelenideTargetLocator.java [hunk 2]
```diff
@@ -156,7 +171,7 @@ public WebDriver window(int index) {
       return Wait().until(windowToBeAvailableAndSwitchToIt(index));
     }
     catch (TimeoutException e) {
-      throw wrapThrowable(e, timeout);
+      throw new NoSuchWindowException("Window with index not found: " + index);
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/SelenideTargetLocator.java]: Don't forget to wrap the original exception: 

```
throw new NoSuchWindowException("Window with index not found: " + index, e);
```

**Comment by tsukakei** [comment]
Thank you for your instruction and review!I fixed

**Comment by rosolko** [comment]
@tsukakei 
Looks good. 
Only one point - can you please assert exception with message also. At least in one test.
Just add `.hasMessage("<message>")` after `.isInstanceOf()` call.

**Comment by asolntsev** [comment]
Merged.

@tsukakei @symonk @rosolko Thank you for your effort!

**Pull request merged**